plugin/nvim-treesitter.vim

```diff
@@ -18,3 +18,23 @@ ts_available_modules = function()
 end
 require'nvim-treesitter'.setup()
 EOF
+
+let g:loaded_nvim_treesitter = 1
+
+" Text objects
+function! s:text_obj_decl(plug_mapping, default_mapping, funcname)
+  let l:formatted_plug = printf('<Plug>(%s)', a:plug_mapping)
+  if empty(hasmapto(l:formatted_plug))
+    execute printf('vmap %s %s', a:default_mapping, l:formatted_plug)
+    execute printf('omap %s %s', a:default_mapping, l:formatted_plug)
+  endif
+
+  execute printf("vnoremap <silent> %s :<C-U>call %s()<CR>", l:formatted_plug, a:funcname)
+  execute printf("omap <silent> %s :normal v%s<CR>", l:formatted_plug, l:formatted_plug)
+endfunction
+
+call s:text_obj_decl('nvim-treesitter-node', 'grn', 'nvim_treesitter#select_node_incr')
+call s:text_obj_decl('nvim-treesitter-context', 'grc','nvim_treesitter#select_scope_incr')
```

Comment:
actually, i think those mappings should only be made on buffer where treesitter is active.  I'm going to test this PR before giving it a go, but i think it might just be fine.